Sources/PackageCollectionsModel/PackageCollectionModel+v1.swift

```diff
@@ -339,3 +338,145 @@ extension PackageCollectionModel.V1.ProductType: Codable {
+                self.issuer = issuer
+            }
+
+            /// Generic certificate name (e.g., subject, issuer)
+            public struct Name: Equatable, Codable {
+                /// Common name
+                public let commonName: String
+
+                /// Creates a `Name`
+                public init(commonName: String) {
+                    self.commonName = commonName
+                }
+            }
+        }
+    }
+}
+
+extension PackageCollectionModel.V1.SignedCollection: Codable {
+    enum CodingKeys: String, CodingKey {
+        // Collection properties
+        case name
+        case overview
+        case keywords
+        case packages
+        case formatVersion
+        case revision
+        case generatedAt
+        case generatedBy
+
+        case signature
+    }
+
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encode(self.collection.name, forKey: .name)
+        try container.encodeIfPresent(self.collection.overview, forKey: .overview)
+        try container.encodeIfPresent(self.collection.keywords, forKey: .keywords)
+        try container.encode(self.collection.packages, forKey: .packages)
+        try container.encode(self.collection.formatVersion, forKey: .formatVersion)
+        try container.encodeIfPresent(self.collection.revision, forKey: .revision)
+        try container.encode(self.collection.generatedAt, forKey: .generatedAt)
+        try container.encodeIfPresent(self.collection.generatedBy, forKey: .generatedBy)
+        try container.encode(self.signature, forKey: .signature)
+    }
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.collection = try PackageCollectionModel.V1.Collection(from: decoder)
+        self.signature = try container.decode(PackageCollectionModel.V1.Signature.self, forKey: .signature)
+    }
+
+    // MARK: - Extract value for single key path
+
+    static let keyPathKey = "key_path"
+
+    public static func collection(from data: Data, using decoder: JSONDecoder) throws -> PackageCollectionModel.V1.Collection {
+        guard let keyPathUserInfoKey = CodingUserInfoKey(rawValue: Self.keyPathKey) else {
+            throw KeyPathValueError.invalidUserInfo
+        }
+        decoder.userInfo[keyPathUserInfoKey] = \PackageCollectionModel.V1.SignedCollection.collection
+        return try decoder.decode(KeyPathValue<PackageCollectionModel.V1.Collection>.self, from: data).value
+    }
+
+    public static func signature(from data: Data, using decoder: JSONDecoder) throws -> PackageCollectionModel.V1.Signature {
+        guard let keyPathUserInfoKey = CodingUserInfoKey(rawValue: Self.keyPathKey) else {
+            throw KeyPathValueError.invalidUserInfo
+        }
+        decoder.userInfo[keyPathUserInfoKey] = \PackageCollectionModel.V1.SignedCollection.signature
```

Comment:
Is the use of a dictionary entry here to support different versions of key signature with the same decoder/encoder?

Reply:
This is for us to decode a specific key (either collection or signature in this case), e.g. we would want to validate the signature before decoding the rest of the bytes. It's looking more like we won't need this code though (i.e., we'd just decode the whole thing), and I will delete it if that's the case.